pom.xml

```diff
@@ -99,7 +106,6 @@
     <dependency>
       <groupId>io.opentracing</groupId>
       <artifactId>opentracing-util</artifactId>
-      <scope>test</scope>
```

Comment:
Please make this _provided_ and _optional=true_ to avoid unnecessary transitive dependencies

Reply:
> It would result in `java.lang.NoClassDefFoundError`.

Currently, yes.
However, as I explained in https://github.com/opentracing-contrib/java-concurrent/pull/12#issuecomment-421972874, the _util_ usage is entirely optional (based on an instanceof check, which can only happen if _the application_ already added the util lib to the classpath).  
Therefore, catching the `NoClassDefFoundError` and interpreting it as "probably isn't an instance of `AutoFinishScopeManager`" is valid logic in my opinion.

Libraries should not declare transitive dependencies if they're optional.

Reply:
That can be quite expensive to have it in runnable/callable.

Reply:
```java
if (tracer.scopeManager() instanceof AutoFinishScopeManager && tracer.scopeManager().active() != null)
       cont = ((AutoFinishScope) tracer.scopeManager().active()).capture();
    else
      cont = null;
```

Could be replaced by:
```java
if (hasAutoFinishScopeManager(tracer) && tracer.scopeManager().active() != null) {
    cont = ((io.opentracing.util.AutoFinishScope) tracer.scopeManager().active()).capture();
}
...
private static boolean hasAutoFinishScopeManager(Tracer tracer) {
    try {
        return tracer.scopeManager() instanceof io.opentracing.util.AutoFinishScopeManager;
    } catch (NoClassDefFoundError utilityJarNotLoaded) {
        LOGGER.finest("The opentracing-util library is not available, therefore the ScopeManager cannot be the AutoFinishScopeManager.");
        return false;
    }
}
```

Just make sure not to _import_ `io.opentrating.util` classes..

I agree this borders on being too far-fetched, just pointing out the possibilities.

Reply:
> That can be quite expensive to have it in runnable/callable.

Ok, then create a boolean constant and populate it in a static code block when the class is loaded..

Reply:
If we have consensus on the last proposal, static boolean. Then I don't mind adding that.